public/app/core/components/AppChrome/AppChromeService.tsx

```diff
@@ -93,14 +100,14 @@ export class AppChromeService {
     return useObservable(this.state, this.state.getValue());
   }
 
-  public onToggleMegaMenu = () => {
-    const isOpen = !this.state.getValue().megaMenuOpen;
-    reportInteraction('grafana_toggle_menu_clicked', { action: isOpen ? 'open' : 'close' });
-    this.update({ megaMenuOpen: isOpen });
-  };
-
-  public setMegaMenu = (megaMenuOpen: boolean) => {
-    this.update({ megaMenuOpen });
+  public setMegaMenu = (newMegaMenuState: AppChromeState['megaMenu']) => {
+    if (config.featureToggles.dockedMegaMenu) {
+      window.localStorage.setItem('grafana.navigation.docked', JSON.stringify(newMegaMenuState === 'docked'));
+      reportInteraction('grafana_mega_menu_state', { state: newMegaMenuState });
+    } else {
+      reportInteraction('grafana_toggle_menu_clicked', { action: newMegaMenuState === 'open' ? 'open' : 'close' });
+    }
+    this.update({ megaMenu: newMegaMenuState });
```

Comment:
If I'm understanding this correctly, at the moment we don't have any interaction reporting?

Why do we report different events depending on whether the feature flag is enabled?

Reply:
ah sorry, i see why it's a bit confusing. we did have interaction reporting as part of `onToggleMegaMenu`, but _toggling_ the menu doesn't really make sense now there's 3 states. so i've removed the `onToggleMegaMenu` method and instead we're just using `setMegaMenu`. so i've added interaction tracking to that method to match what was in `onToggleMegaMenu`.

in terms of the difference between when the toggle is on and off, it's just easier to report the new state itself than create these string actions which don't directly relate to anything in code.